shared/security/src/main/java/io/pravega/shared/security/token/JsonWebToken.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) 2019 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.security.token;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.MalformedJwtException;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.SignatureException;
+import io.pravega.auth.InvalidTokenException;
+import io.pravega.auth.TokenException;
+import io.pravega.auth.TokenExpiredException;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import lombok.NonNull;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Creates Json Web Tokens (JWT). Also, parses a JWT.
+ */
+@ToString
+@Slf4j
+public class JsonWebToken {
```

Comment:
This class is not thread safe. Either make all fields final or use some synchronization method to resolve this.

Reply:
I've made all the fields final. Thanks.